File: android/src/main/java/io/inway/ringtone/player/AlarmNotificationMeta.java
package io.inway.ringtone.player;

import java.io.Serializable;
import java.util.Map;

public class AlarmNotificationMeta implements Serializable {
    private final Map<String, Object> notificationMetaValues;

    public AlarmNotificationMeta(Map<String, Object> notificationMetaValues) {
        this.notificationMetaValues = notificationMetaValues;
    }

    public CharSequence getContentTitle() {
        return (CharSequence) notificationMetaValues.get("contentTitle");
    }

    public CharSequence getContentText() {
        return (CharSequence) notificationMetaValues.get("contentText");
    }

    public CharSequence getSubText() {
        return (CharSequence) notificationMetaValues.get("subText");
    }

    public String getIconDrawableResourceName() {
        return (String) notificationMetaValues.get("iconDrawableResourceName");
    }

    public String getActivityClassLaunchedByIntent() {
        return (String) notificationMetaValues.get("activityClassLaunchedByIntent");
    }
}

File: lib/alarm_notification_meta.dart
class AlarmNotificationMeta {
  final String activityClassLaunchedByIntent;
  final String iconDrawableResourceName;
  final String contentTitle;
  final String contentText;
  final String subText;

  AlarmNotificationMeta(this.activityClassLaunchedByIntent, this.iconDrawableResourceName,
      {this.contentTitle, this.contentText, this.subText});

  Map<String, dynamic> toMap() => {
        'activityClassLaunchedByIntent': activityClassLaunchedByIntent,
        'iconDrawableResourceName': iconDrawableResourceName,
        'contentTitle': contentTitle,
        'contentText': contentText,
        'subText': subText,
      };
}
